js/src/gtag-events/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { addAction } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { namespace, actionPrefix } from './constants';
+import {
+	getProductObject,
+	retrievedVariation,
+	trackAddToCartEvent,
+} from './utils';
+
+/* global jQuery */
+
+addAction(
+	`${ actionPrefix }-cart-add-item`,
+	namespace,
+	( { product, quantity = 1 } ) => {
+		trackAddToCartEvent( product, quantity );
+	}
+);
+
+const addToCartClick = function ( event ) {
+	const data = event.target.dataset;
+	const product = getProductObject( { id: data.product_id } );
+
+	trackAddToCartEvent( product, data.quantity || 1 );
+};
+
+const singleAddToCartClick = function ( event ) {
+	const cartForm = event.target.closest( 'form.cart' );
+	if ( ! cartForm ) {
+		return;
+	}
+
+	const addToCart = cartForm.querySelector( '[name=add-to-cart]' );
+	if ( ! addToCart ) {
+		return;
+	}
+
+	const variationId = cartForm.querySelector( '[name=variation_id]' );
+	const quantity = cartForm.querySelector( '[name=quantity]' );
+
+	const product = getProductObject( {
+		id: parseInt( variationId ? variationId.value : addToCart.value, 10 ),
+	} );
+
+	trackAddToCartEvent(
+		product,
+		quantity ? parseInt( quantity.value, 10 ) : 1
+	);
+};
+
+// Register for add_to_cart click events.
+window.onload = function () {
+	document
+		.querySelectorAll(
+			'.add_to_cart_button:not( .product_type_variable, .product_type_grouped )'
+		)
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', addToCartClick );
+		} );
+
+	document
+		.querySelectorAll( '.single_add_to_cart_button' )
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', singleAddToCartClick );
+		} );
+};
+
+// Register for jQuery event to update product data.
+if ( typeof jQuery === 'function' ) {
+	jQuery( document ).on(
+		'found_variation',
+		'form.cart',
+		function ( event, variation ) {
+			retrievedVariation( variation );
+		}
+	);
```

Comment:
❓ 
I take as granted that 'found_variation' is triggered by `jQuery.trigger()`. Otherwise I'd suggest to use native `addEventListener` instead

Reply:
Yes this is triggered by jQuery. I tried to depend on jQuery as little as possible, but until WooCommerce rewrites it's JS on product pages this the balance we need to maintain to remain compatible.